### PR TITLE
Bump versions of dependencies

### PR DESCRIPTION
**Pull request by ots22**
This PR relaxes the required version of several dependencies, removes an upper bound on the version of python required and upgrades Bokeh to version 3 (which makes several interface changes so this is now the minimum required version).  Some other changes following the upgraded Bokeh version: the mode of figure sizing (the previous mode results in a figure of zero height in the notebook) and using y-restricted box selection which is now available.

The PR also removes the poetry lockfile. Installing with poetry results in an unusual set of dependencies being selected for recent versions of python that we do not want to pin for development.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -25,21 +25,22 @@ classifiers = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.12"
-bokeh = "^2.3"
-numpy = "^1.21"
-pandas = "^1.3"
-pydantic = "^1.8"
+python = ">=3.8"
+bokeh = "^3.0"
```
Comment (layik) [inline pyproject.toml:29]: The only suggestion I have is to fix this to 3.4 and keep this updated as the package grows.

**Review by layik — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (ots22) [reply pyproject.toml]: I've checked with the latest of each 3.* version of Bokeh: 3.1 seems to be the minimum required so I've updated to this.  An fresh install or update to the package should use the latest Bokeh.

**Comment by layik** [comment]
Hi @ots22. Thank you. I just ran another test with "bokeh 3.4" and works fine. Lets just make sure we tell users this and I am happy to merge this. Amazing effort!

**Comment by ots22** [comment]
Thanks for the review @layik. Just to note that I don't think there is any particular requirement for Bokeh 3.4 (it seems to work with anything >3.1).  I did have some trouble with missing figures when repeatedly installing/upgrading for testing purposes (which I think you also encountered), but I'm now fairly sure this was caused by browser caching rather than a particular Bokeh version not working. I was about to make an issue - it might be worth a note in the docs.

**Comment by layik** [comment]
Fantastic, let's merge and release. Let me know if I can help. Thank you

**Comment by ots22** [comment]
@layik - I think you were right about needing Bokeh 3.4 in fact, some notes in #134 

We now require 3.4 and therefore python 3.9.

**Comment by layik** [comment]
Thanks @ots22 :)

**Pull request merged**